tests/conftest.py

```diff
@@ -86,6 +114,36 @@ async def echo_sse_server() -> AsyncGenerator[dict[str, str], None]:
         await process.wait()
 
 
+@pytest.fixture(scope="session")
+async def date_streamable_http_server(worker_id: str) -> AsyncGenerator[dict[str, Any]]:
+    """This fixture runs a FastMCP server in a subprocess.
+    I thought about trying to mock all the individual mcp client calls,
+    but I went with this because this way we don't need to actually mock anything.
+    This is similar to what MCPAdapt does in their testing https://github.com/grll/mcpadapt/blob/main/tests/test_core.py
+    """
+    import asyncio
+
+    port = 19010
```

Comment:
Again, issues with server ports, this time due to xdist setting up several processes. Possible alternative approaches would be:

1. Moving to in-place async servers instead of subprocess, and bind to random ports.
1. Grouping tests to ensure some fixtures are only spawned in a single worker, where needed.